beprof/curve.py

```diff
@@ -66,19 +66,74 @@ def y_at_x(self, x):
             return x
         return np.interp(x, self.x, self.y, left=np.nan, right=np.nan)
 
+    def change_domain(self, domain):
+
+        # check if new domain includes in the orginal domain
+        if np.max(domain) > np.max(self.x):
+            # maybe some exception here or stderr log ?
+            # these prints only for testing purpose
+            print('Error1')
+            print('domain.max: ', np.max(domain), '|| self.max: ', np.max(self.x))
+            return self
+        if np.min(domain) < np.min(self.x):
+            print('Error2')
+            print('domain.min: ', np.min(domain), '|| self.min: ', np.min(self.x))
+            return self
+        y = np.interp(domain, self.x, self.y)
+        obj = Curve([[domain[ind], y[ind]] for ind in range(0, len(y))])
+        return obj
+
+    def fixed_step_domain(self, step=0.1, fixp=0):
+        section = (np.min(self.x), np.max(self.x))
+        count_start = (abs(fixp - section[0]) / step)
+        count_stop = (abs(fixp - section[1]) / step)
+
+        # depending on position of fixp with respect to the orginal domain
+        # may be 1 of 3 cases:
+
+        if fixp < section[0]:
+            count_start = int(np.ceil(count_start))
+            count_stop = int(np.floor(count_stop))  # floor is not necessary, just makes the code
+        elif fixp > section[1]:                     # and my intentions more clear
+            count_start = -int(np.floor(count_start))
```

Comment:
I don't think it'll help you here, but might be useful in other situations:
http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.arange.html
Two important fragments:
> When using a non-integer step, such as 0.1, the results will often not be consistent. It is better to use linspace for these cases.
> For floating point arguments, the length of the result is ceil((stop - start)/step). Because of floating point overflow, this rule may result in the last element of out being greater than stop.